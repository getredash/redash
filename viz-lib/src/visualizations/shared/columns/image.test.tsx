import React from "react";
import enzyme from "enzyme";

import Column from "./image";

function findByTestID(wrapper: any, testId: any) {
  return wrapper.find(`[data-test="${testId}"]`);
}

function mount(column: any, done: any) {
  return enzyme.mount(
    <Column.Editor
      // @ts-expect-error ts-migrate(2322) FIXME: Type '{ visualizationName: string; column: any; on... Remove this comment to see the full error message
      visualizationName="Test"
      column={column}
      onChange={changedColumn => {
        expect(changedColumn).toMatchSnapshot();
        done();
      }}
    />
  );
}

describe("Visualizations -> Table -> Columns -> Image", () => {
  describe("Editor", () => {
    test("Changes URL template", done => {
      const el = mount(
        {
          name: "a",
          imageUrlTemplate: "{{ @ }}",
        },
        done
      );

      findByTestID(el, "Table.ColumnEditor.Image.UrlTemplate")
        .last()
        .find("input")
        .simulate("change", { target: { value: "http://{{ @ }}.jpeg" } });
    });

    test("Changes width", done => {
      const el = mount(
        {
          name: "a",
          imageWidth: null,
        },
        done
      );

      findByTestID(el, "Table.ColumnEditor.Image.Width")
        .last()
        .find("input")
        .simulate("change", { target: { value: "400" } });
    });

    test("Changes height", done => {
      const el = mount(
        {
          name: "a",
          imageHeight: null,
        },
        done
      );

      findByTestID(el, "Table.ColumnEditor.Image.Height")
        .last()
        .find("input")
        .simulate("change", { target: { value: "300" } });
    });

    test("Changes title template", done => {
      const el = mount(
        {
          name: "a",
          imageUrlTemplate: "{{ @ }}",
        },
        done
      );

      findByTestID(el, "Table.ColumnEditor.Image.TitleTemplate")
        .last()
        .find("input")
        .simulate("change", { target: { value: "Image {{ @ }}" } });
    });
  });
});
