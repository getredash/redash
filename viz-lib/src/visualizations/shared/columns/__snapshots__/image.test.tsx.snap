// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visualizations -> Table -> Columns -> Image Editor Changes URL template 1`] = `
Object {
  "imageUrlTemplate": "http://{{ @ }}.jpeg",
}
`;

exports[`Visualizations -> Table -> Columns -> Image Editor Changes height 1`] = `
Object {
  "imageHeight": "300",
}
`;

exports[`Visualizations -> Table -> Columns -> Image Editor Changes title template 1`] = `
Object {
  "imageTitleTemplate": "Image {{ @ }}",
}
`;

exports[`Visualizations -> Table -> Columns -> Image Editor Changes width 1`] = `
Object {
  "imageWidth": "400",
}
`;
